### DISCUSSION TITLE
Question about set_data api when using persistent dataset.

### DISCUSSION THREAD

**Discussion by JiaxinZhuang** [Project-MONAI/MONAI#6742] [Q&A] [answered]
Hi, I have read the parts about the persistent datasets. I want to know when should we use set_data?
Let me be more specific. If we train a model for 100 epochs with a persistent dataset (with random operations like rand crops), should we call set_data in some epoch?

**Comment by KumoLiu** [comment, marked as answer]
Hi @JiaxinZhuang, you don't need to call `set_data` since `PersistentDataset` will only cache deterministic transforms. You may need to call it only if you want to reset the input data and delete all the outdated cache content.

Hope it helps, thanks!

**Reply by JiaxinZhuang** [reply]
Thanks for the reply.
